src/test/java/org/apache/maven/shared/dependency/analyzer/asm/ASMDependencyAnalyzerTest.java

```diff
@@ -40,4 +43,12 @@ void test() throws Exception {
 
         assertThat(result).isNotEmpty();
     }
+
+    @Test
+    void verify_implicit_class_reference_included_in_used_classes() throws IOException {
```

Comment:
naming convention seems wonky here

Reply:
Snake case is a habit, I find it more readable. I have no strong opinions. If it should be changed to camelCase then let me know and I will update.

Please also check if my logic is sound. I am no expert in ASM and byte code. Naming is easily fixed.